src/main/native/smithwaterman/IntelSmithWaterman.cc

```diff
@@ -87,13 +87,16 @@ JNIEXPORT jint JNICALL Java_com_intel_gkl_smithwaterman_IntelSmithWaterman_align
     env->ReleasePrimitiveArrayCritical(alt, alternate, 0);
     env->ReleasePrimitiveArrayCritical(cigar, cigarArray, 0);
 
-    if(offset == -1 || env->ExceptionCheck()) {
+    if(offset == -1) {
         env->ExceptionClear();
         env->ThrowNew(env->FindClass("java/lang/IllegalArgumentException"), "Invalid offset value");
+        return -1;
     }
 
     if(offset == -2) {
+        env->ExceptionClear();
         env->ThrowNew(env->FindClass("java/lang/OutOfMemoryError"), "Memory allocation issue");
+        return -1;
     }
```

Comment:
Those conditions are wrong and they will be removed in my next patch. The offset values set to -1 or -2 are perfectly correct. I don't know why anyone put those conditions here. They break Smith-Waterman.

Reply:
The out of Memory Error condition needs to be passed back to the java layer from C.   -2 return code was used as a way to propagate the error but is unsafe.  A new way is needed to pass the out of memory error.  There is now a max value input check in Java so the -1 condition  needs to be reworked and maybe removed.